src/hugr.rs

```diff
@@ -447,4 +467,55 @@ mod test {
         let hugr = simple_dfg_hugr();
         assert_matches!(hugr.get_io(hugr.root()), Some(_));
     }
+
+    #[test]
+    fn extension_instantiation() -> Result<(), Box<dyn Error>> {
+        const BIT: Type = crate::extension::prelude::USIZE_T;
+        let r = ExtensionSet::singleton(&"R".into());
+
+        let root = NodeType::pure(ops::DFG {
```

Comment:
driveby nit: `NodeType::pure` seems a bit misnamed. The so-called "purity" is not about the nodetype - it's about what's already present on the inputs before we even get to that node. A "pure" nodetype would be one with an empty delta; `pure` should be called `new_no_extensions` or something...."no extensions" == unextended(??) with synonyms "brief",  "truncated", "compact" and "concise". I admit none of these are great. "Basic"? "Retracted" (!)?? Maybe just `NodeType::no_extensions` paralleling `open_extensions` ?

Reply:
That's fair, but I'm not sure what the better name is. Maybe `open_extensions => open` and `pure => closed`?

Reply:
Yes - let's leave renaming to another PR, but I see two reasonable schemes: `open_extensions` and `no_extensions` or `open` and `closed` (where currently we have `open_extensions` and `pure`)